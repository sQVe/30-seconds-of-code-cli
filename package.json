{
  "name": "30s",
  "version": "0.1.2",
  "description": "A command-line application for 30 seconds of code snippets",
  "author": "Oskar Grunning <oskargrunning@gmail.com>",
  "license": "CC0-1.0",
  "homepage": "https://github.com/sQVe/30s#30s",
  "repository": "github:sQVe/30s",
  "bugs": "https://github.com/sQVe/30s/issues",
  "keywords": [
    "javascript",
    "snippets",
    "cli"
  ],
  "scripts": {
    "build": "rm -rf lib && npm run build:js && npm run build:parse",
    "build:copy": "cp src/snippets.json lib/",
    "build:js": "NODE_ENV=production babel src --out-dir lib",
    "build:parse": "node lib/parser.js",
    "format": "prettier-eslint --write '{**/*,*}.{js,json,jsx,md,yaml}'",
    "format:check": "prettier-eslint --list-different '{**/*,*}.{js,json,jsx,md,yaml}'",
    "lint": "eslint '{**/*,*}.{js,jsx}'",
    "prepare": "npm run build",
    "start": "babel-node src/index.js",
    "test": "jest",
    "test:cov": "jest --coverage",
    "test:cov:up": "jest --coverage --coverageReporters=text-lcov | coveralls",
    "test:integration": "jest test/integration --testPathIgnorePatterns",
    "test:watch": "jest --watch --onlyChanged"
  },
  "engines": {
    "node": ">= 8"
  },
  "bin": {
    "30s": "./lib/index.js"
  },
  "files": [
    "lib/"
  ],
  "devDependencies": {
    "@babel/cli": "~7.2.0",
    "@babel/core": "~7.2.0",
    "@babel/node": "~7.2.0",
    "@babel/preset-env": "~7.3.0",
    "@types/jest": "^24.0.0",
    "@types/node": "^10.12.0",
    "@types/ramda": "^0.25.40",
    "babel-core": "7.0.0-bridge.0",
    "babel-eslint": "~10.0.0",
    "babel-jest": "~24.1.0",
    "coveralls": "~3.0.0",
    "eslint": "~5.12.0",
    "eslint-config-sqve": "~2.1.0",
    "eslint-config-standard": "~12.0.0",
    "eslint-config-standard-react": "~7.0.0",
    "eslint-plugin-fp": "~2.3.0",
    "eslint-plugin-import": "~2.15.0",
    "eslint-plugin-node": "~8.0.0",
    "eslint-plugin-promise": "~4.0.0",
    "eslint-plugin-react": "~7.12.0",
    "eslint-plugin-standard": "~4.0.0",
    "husky": "~1.3.0",
    "jest": "~24.1.0",
    "lint-staged": "~8.1.0",
    "prettier": "~1.16.0",
    "prettier-eslint-cli": "~4.7.0",
    "semantic-release": "~15.13.0"
  },
  "dependencies": {
    "@babel/polyfill": "~7.2.0",
    "chalk": "~2.4.0",
    "cli-highlight": "~2.0.0",
    "clipboardy": "~1.2.0",
    "commander": "~2.19.0",
    "marked": "~0.6.0",
    "marked-terminal": "~3.2.0",
    "ramda": "~0.26.0"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "*.{js,jsx}": [
      "prettier-eslint --write",
      "eslint",
      "jest --findRelatedTests",
      "git add"
    ],
    "*.json": [
      "prettier-eslint --write",
      "git add"
    ]
  }
}
