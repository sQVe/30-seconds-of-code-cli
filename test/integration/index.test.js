import path from 'path'
import { spawnSync } from 'child_process'

import { version } from '../../package.json'

const cliPath = path.resolve(__dirname, '../../lib/30s.js')

const callCli = (...args) =>
  spawnSync('node', [cliPath, ...args])
    .stdout.toString()
    .trim()
const commands = ['search', 'view', 'tag']

describe('Cli', () => {
  describe('version option', () => {
    it('should print the package version', async () => {
      expect(callCli('-V')).toEqual(version)
      expect(callCli('--version')).toEqual(version)
    })
  })

  describe('help option', () => {
    it('should print help page', async () => {
      expect(callCli('-h')).toMatchSnapshot()
      expect(callCli('--help')).toMatchSnapshot()
    })

    commands.forEach(command => {
      it(`should print help page for ${command} command`, () => {
        expect(callCli(command, '-h')).toMatchSnapshot()
        expect(callCli(command, '--help')).toMatchSnapshot()
      })
    })
  })

  describe('input safe guard', () => {
    // TODO: Write this.
  })

  commands.forEach(command => {
    describe(`${command} command`, () => {
      // TODO: Write correct it cases here.

      it(`should output test strings`, async () => {
        expect(callCli(command, 'foo')).toEqual(
          JSON.stringify([command, 'foo', false, false, 'itced'])
        )
        expect(callCli(command, 'foo', '--layout', 'ic')).toEqual(
          JSON.stringify([command, 'foo', false, false, 'ic'])
        )
        expect(callCli(command, 'foo', '--json')).toEqual(
          JSON.stringify([command, 'foo', false, true, 'itced'])
        )
        expect(callCli(command, 'foo', '--cp')).toEqual(
          JSON.stringify([command, 'foo', true, false, 'itced'])
        )
      })
    })
  })
})
