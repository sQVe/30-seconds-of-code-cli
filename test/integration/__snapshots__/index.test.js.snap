// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cli help option should print help page 1`] = `
"Usage: 30s [options] [command]

Options:
  -V, --version               output the version number
  -h, --help                  output usage information

Commands:
  r|random [options]          view random snippet
  s|search [options] [query]  fuzzy search snippets by id
  t|tag [options] [id]        view snippets by tag
  v|view [options] [id]       view snippet with id
  * [id]                      view snippet with id

Examples:
  30s head
  30s v head
  30s view head

  30s r
  30s random

  30s s -j flatten
  30s search --json flatten

  30s t -l ic array
  30s tag --layout ic array"
`;

exports[`Cli help option should print help page 2`] = `
"Usage: 30s [options] [command]

Options:
  -V, --version               output the version number
  -h, --help                  output usage information

Commands:
  r|random [options]          view random snippet
  s|search [options] [query]  fuzzy search snippets by id
  t|tag [options] [id]        view snippets by tag
  v|view [options] [id]       view snippet with id
  * [id]                      view snippet with id

Examples:
  30s head
  30s v head
  30s view head

  30s r
  30s random

  30s s -j flatten
  30s search --json flatten

  30s t -l ic array
  30s tag --layout ic array"
`;

exports[`Cli help option should print help page for random command 1`] = `
"Usage: 30s r|random [options]

view random snippet

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for random command 2`] = `
"Usage: 30s r|random [options]

view random snippet

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for search command 1`] = `
"Usage: 30s s|search [options] [query]

fuzzy search snippets by id

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for search command 2`] = `
"Usage: 30s s|search [options] [query]

fuzzy search snippets by id

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for tag command 1`] = `
"Usage: 30s t|tag [options] [id]

view snippets by tag

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for tag command 2`] = `
"Usage: 30s t|tag [options] [id]

view snippets by tag

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for view command 1`] = `
"Usage: 30s v|view [options] [id]

view snippet with id

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for view command 2`] = `
"Usage: 30s v|view [options] [id]

view snippet with id

Options:
  -c, --cp               copy code to clipboard (default: false)
  -j, --json             output in json format (default: false)
  -l, --layout <layout>  print in specified layout (default: \\"itced\\")
  -h, --help             output usage information"
`;
